cmd/deploy/remote.go

```diff
@@ -169,6 +180,20 @@ func (rd *remoteDeployCommand) deploy(ctx context.Context, deployOptions *Option
 		fmt.Sprintf("%s=%s", constants.OktetoGitCommitEnvVar, os.Getenv(constants.OktetoGitCommitEnvVar)),
 		fmt.Sprintf("%s=%d", constants.OktetoInvalidateCacheEnvVar, int(randomNumber.Int64())),
 	)
+
+	sshSock := os.Getenv(rd.sshAuthSockEnvvar)
+	if sshSock == "" {
+		sshSock = os.Getenv("SSH_AUTH_SOCK")
+	}
+
+	if sshSock != "" {
+		sshSession := types.BuildSshSession{Id: "remote", Target: sshSock}
+		buildOptions.SshSessions = append(buildOptions.SshSessions, sshSession)
+		buildOptions.Secrets = append(buildOptions.Secrets, fmt.Sprintf("id=known_hosts,src=%s/.ssh/known_hosts", home))
```

Comment:
does this work in windows? I think we we have `--remote` e2e tests, it would be great to have an e2e test covering this scenario in linux/window
I haven't tried, but I think the path is slightly different:
https://superuser.com/questions/311886/where-is-the-known-hosts-file-for-openssh-for-windows

Reply:
another option, if this approach gets too complicated, is to use a static file. and the user can always update their own known host file to the deploy `image` if needed
or infer the file in a similar way as we do in the installer:
https://github.com/okteto/pipeline-installer-image/blob/b0d72026c80eb8947adfc9c5b4f4f47a2f7bca8a/cmd/git/ssh.go#L53

Reply:
Yeah, this was incorrect and started failing in windows once I updated the test to `filepath.Join(home, ".ssh", "known_hosts")`. It should be fixed now.

Reply:
I was trying to create an integration test but it's tricky since it requires credentials or a private repo to test an ssh connection from inside the deploy. Adding these credentials as envvars would make `go test github.com/okteto/okteto/integration/... -tags="common integration actions"`  stop working out of the box. Is there a recommended way or do we have something in place for doing this? cc @okteto/lake.